services/tasks/local.go

```diff
@@ -259,36 +259,50 @@ func (l *local) DeleteProcess(ctx context.Context, r *api.DeleteProcessRequest,
 }
 
 func processFromContainerd(ctx context.Context, p runtime.Process) (*task.Process, error) {
-	state, err := p.State(ctx)
-	if err != nil {
+	ch := make(chan *task.Process, 1)
+	errCh := make(chan error, 1)
+	go func() {
+		state, err := p.State(ctx)
```

Comment:
Should we add a timeout to this context to prevent it from blocking if `State` never returns?